FBReader/src/main/java/org/geometerplus/android/util/UIUtil.java

```diff
@@ -64,8 +66,13 @@ public static void wait(String key, Runnable action, Context context) {
 			final String message =
 				ZLResource.resource("dialog").getResource("waitMessage").getResource(key).getValue();
 			ourTaskQueue.offer(new Pair(action, message));
-			if (ourProgress == null) {
-				ourProgress = ProgressDialog.show(context, null, message, true, false);
+			if (ourProgress == null ) {
```

Comment:
This if can be inverted to be if (oP != null) return,  this way we remove the else and the nested try catch.